### PR TITLE
Compile Sass with JavaScript package

### PR DESCRIPTION
**Pull request by aduth**
This changes the Sass compilation to use the `@18f/identity-build-sass` package published through `18f/identity-idp`.

Refer to https://github.com/18F/identity-site/pull/1000 and https://github.com/18F/identity-site/pull/1003 for rationale.

Additionally:

- Brings the developer site compilation into closer alignment with other Login.gov products (i.e. the brochure site).
- Resolves some build failures observed in Cloud.gov Pages related to Sass compilation, especially with the introduction of the Login.gov Design System required Sass modules in #319 (["Jekyll::Converters::Scss encountered an error [...] String can't be coerced into Integer"](https://pages.cloud.gov/sites/28/builds/298326/logs)).
- Allows us to minify Sass compiled output via built-in `NODE_ENV` handling

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,15 +26,21 @@ Run tests
 docker-compose run web make test
 ```
 
-You can drop into a bash shell in the container with 
+You can drop into a bash shell in the container with
 
 ```sh
 docker-compose run web bash
 ```
```
Comment (zachmargolis) [inline README.md:33]: oh wow I forgot we had docker instructions here, are these still valid? should we remove? (maybe separate PR)

**Review by zachmargolis — APPROVED** [review]
LGTM

File: assets/scss/main.css.scss [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }
+}
+
+#nav-links {
+  width: fit-content !important;
+}
+
+//override for touchpoints survey included via script
+#fba-modal-dialog .usa-banner__header-text {
```
[the remark this answers is not in the document]
Comment (aduth) [reply assets/scss/main.css.scss:24]: We did in #318, yes, with caveat that it "will likely return the survey after tax season". I'd be inclined to remove it as unused though. I can drop a comment on the old pull request about the removal to help with a potential future revert.

File: README.md (hunk 1 above)
Comment (aduth) [reply README.md:33]: > are these still valid? should we remove?

No idea, on either point 😅 The `docker-compose.yml` looks relatively straight-forward, so if it was working at some point, I assume (with low confidence) that it probably wouldn't be impacted by these changes.

**Pull request merged**